tests/unit/CollectionConstructorsTest.php

```diff
@@ -0,0 +1,334 @@
+        self::assertInstanceOf(Collection::class, $test4);
+        self::assertIdenticalIterable(
+            range(1, 5),
+            $test4
+        );
+
+        $classWithStaticMethod = new class() {
+            public static function getValues(): Generator
+            {
+                yield from range(1, 5);
+            }
+        };
+        $test5 = Collection::fromCallable([$classWithStaticMethod, 'getValues']);
+        self::assertInstanceOf(Collection::class, $test5);
+        self::assertIdenticalIterable(
+            range(1, 5),
+            $test5
+        );
+
+        $invokableClass = new class() {
+            public function __invoke(): Generator
+            {
+                yield from range(1, 5);
+            }
+        };
+        $test6 = Collection::fromCallable($invokableClass);
+        self::assertInstanceOf(Collection::class, $test6);
+        self::assertIdenticalIterable(
+            range(1, 5),
+            $test6
+        );
+    }
+
+    public function testFromEmptyConstructor(): void
+    {
+        self::assertIdenticalIterable(
+            [],
+            Collection::empty()
+        );
+    }
+
+    public function testFromFileConstructor()
+    {
+        self::assertIdenticalIterable(
+            ['a', 'b', 'c'],
+            Collection::fromFile(__DIR__ . '/../fixtures/sample.txt')
+        );
+    }
+
+    public function testFromGeneratorConstructor()
+    {
+        $generator = static function () {
+            yield 'a';
+
+            yield 'b';
+
+            yield 'c';
+
+            yield 'd';
+
+            yield 'e';
+        };
+
+        self::assertIdenticalIterable(
+            range('a', 'e'),
+            Collection::fromGenerator($generator())
+        );
+
+        $generator = (static function () {
+            yield 'a';
+
+            yield 'b';
+
+            yield 'c';
+
+            yield 'd';
+
+            yield 'e';
+        })();
+
+        $generator->next();
+        $generator->next();
+
+        self::assertIdenticalIterable(
+            [2 => 'c', 3 => 'd', 4 => 'e'],
+            Collection::fromGenerator($generator)
+        );
+    }
+
+    public function testFromIterableConstructor(): void
+    {
+        self::assertIdenticalIterable(
+            ['A', 'B', 'C'],
+            Collection::fromIterable(range('A', 'C'))
+        );
+
+        $generator = (static function () {
+            yield 'a';
+
+            yield 'b';
+
+            yield 'c';
+
+            yield 'd';
+
+            yield 'e';
+        })();
+
+        $generator->next();
+        $generator->next();
+
+        self::assertIdenticalIterable(
+            [2 => 'c', 3 => 'd', 4 => 'e'],
+            Collection::fromIterable($generator)
+        );
+
+        // TODO: Reimplement with PHPUNIT.
+        /*
+        $subject
+            ->shouldThrow(Error::class)
+            ->during('count');
+         */
```

Comment:
did you forget about these or are they for a future PR?

Reply:
Fixed :)